catowseriOS/catowser/BrowserContent/WebView/WebViewAuthChallengeHandler.swift

```diff
@@ -35,10 +35,12 @@ final class WebViewAuthChallengeHandler {
         queue = DispatchQueue(label: .queueNameWith(suffix: "webview.auth-challenge"))
     }
 
-    func solve(completion: @escaping (Bool?) -> Void) {
+    func solve(completion: @MainActor @Sendable @escaping (Bool?) -> Void) {
         guard challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodServerTrust else {
-            queue.async { [weak self] in
-                self?.completionHandler(.performDefaultHandling, nil)
+            queue.async {
```

Comment:
Try to use MainActor.run although there was an error to not run some code on main